lib/gollum/public/gollum/javascript/gollum.js.erb

```diff
@@ -658,6 +658,54 @@ $(document).ready(function() {
         }
       });
     }
+
+    if (navigator.clipboard) {
+      // Create copy to clipboard buttons for code-blocks
+      const codeBlocks = document.querySelectorAll("pre > code");
+      const iconCopyDefault = `<%=rocticon('copy', {:width => 18, :height => 18})%>`;
+      const iconCopyActive = `<%=rocticon('check', {:width => 18, :height => 18})%>`;
+
+      const copyToClipboard = async (event) => {
+        const { currentTarget: button } = event;
+
+        const codeBlock = document.querySelector(`#${button.dataset.codeblock}`);
+
+        if (!codeBlock) {
+          throw('Error while coping to clipboard: codeblock not found.');
+          return;
+        }
+
+        await navigator.clipboard.writeText(codeBlock.textContent);
+
+        button.innerHTML = iconCopyActive;
+        button.classList.add("tooltipped", "tooltipped-no-delay", "tooltipped-w");
+        button.setAttribute('aria-label', 'Copied!');
+
+        // wait 2 secconds
+        await new Promise(resolve => setTimeout(resolve, 2000));
+
+        button.innerHTML = iconCopyDefault;
+        button.classList.remove("tooltipped", "tooltipped-no-delay", "tooltipped-w");
+        button.setAttribute('aria-label', 'copy to clipboard');
+      };
+
+      codeBlocks.forEach((item, idx) => {
+        // assign a random & unique ID so the element can be selected later.
```

Comment:
This comment is no longer apt, it should just read `assign a unique ID [...]`.

Reply:
Fixed!